core/src/main/java/org/infinispan/stats/impl/CacheContainerStatsImpl.java

```diff
@@ -64,7 +64,11 @@ public void resetStatistics() {
       if (getStatisticsEnabled()) {
          for (String cn : cm.getCacheNames()) {
             if (cm.cacheExists(cn)) {
-               cm.getCache(cn).getAdvancedCache().getStats().reset();
+               try {
+                  cm.getCache(cn).getAdvancedCache().getStats().reset();
+               } catch (SecurityException e) {
+                  // no-op, skipping this cache
```

Comment:
sorry, @tristantarrant , I probably don't understand, so what the correct solution should look like?

> a user has no way of knowing which caches would be included / excluded

exactly, this was original motivation why I started this PR

>  I think also that, when security is enabled, only a cache ADMIN would have access to stats ops anyway.

this is obviously not true, at least in current implementation. Statistics is touched in various places - see linked jira. As an ADMIN of the cache, I get exception when accessing the cache via CLI, as statistics try to update all caches, including internal caches, in this cache script cache and fails as I hadn't `__scriptmanager` role. 

It's also possible we should change statistics implementation (haven't looked on statistics yet, so I have currently no opinion on it), but if we want to iterate over all caches in statistics, I don't see any other solution that skip caches which I haven't access to (or skip security check at all)